cpp/tests/groupby/hash/single_column_max.cu

```diff
@@ -0,0 +1,354 @@
+/*
```

Comment:
Is it possible to reduce duplication in these tests by parameterizing the operator? I know you can't do a cartesian product in gtest, but you can loop over the operators inside each test...

Reply:
I did originally try this, but it had some big problems:
1. The expected output is different for each operator, so you'd have to reimplement groupby on the CPU (with all flavors of null support) in order to compute a reference solution. 
2. You'd have a massive typed-parameterized test list because you want to test a variety of key/value type combinations for each operator. You'd likely end up hitting the max limit of GTest's type parameterized tests, and the test file would take forever to compile. 

It may be worthwhile to eventually revisit 1., especially when adding multi-column key/value tests, but it's a non-trivial effort because you essentially have to re-implement my algorithm on the CPU.